docs/guides/run-elasticsearch-local.rst

```diff
@@ -0,0 +1,99 @@
+==========================================
+Enabling Elasticsearch on the local server
+==========================================
+
+Read the Docs has been using Elasticsearch which is a platform for distributed search and analysis of data in real time. To enable the search feature on your local installation, you need to install the elasticsearch locally and run the Elastic server. 
+
+Installation has been mainly divided into following steps.
+
+1. Installing Java
+------------------
+
+Elasticsearch requires JAVA 8 or later. Use `Oracle official documentation <http://www.oracle.com/technetwork/java/javase/downloads/index.html>`_. 
+or opensource distribution like `OpenJDK <http://openjdk.java.net/install/>`_.
+
+After installing java, verify the installation by,::
+
+	$ java -version
+
+The result should be something like this::
+
+	openjdk version "1.8.0_151"
+	OpenJDK Runtime Environment (build 1.8.0_151-8u151-b12-0ubuntu0.16.04.2-b12)
+	OpenJDK 64-Bit Server VM (build 25.151-b12, mixed mode)
+
+
+2. Downloading and installing Elasticsearch
+-------------------------------------------
+
+Elasticsearch can be downloaded directly from elastic.co in zip, tar.gz, deb, or rpm packages. For Ubuntu, it's best to use the deb (Debian) package which will install everything you need to run Elasticsearch.
+
+RTD currently uses elasticsearch 1.x which can be easily downloaded and installed from the `Official Website 
+<https://www.elastic.co/downloads/>`_.
+
+	1. First download the deb package version 1.3.8 from this `Link <https://www.elastic.co/downloads/past-releases/elasticsearch-1-3-8>`_.
```

Comment:
I'd merge the first item with the paragraph above. Since we need to install a specific version of Elasticsearch, we don't need to mention the generic downloads page. We can just link to the download page of 1.3.8.